build/BuildInfo.targets

```diff
@@ -0,0 +1,40 @@
+<Project ToolsVersion="15.0">
+  <Target Name="WriteBuildInfoProps"
+          DependsOnTargets="BuildDotnetCliBuildFramework">
+
+    <GetCurrentRuntimeInformation>
+      <Output TaskParameter="Rid" PropertyName="HostRid" />
+      <Output TaskParameter="OSName" PropertyName="HostOSName" />
+    </GetCurrentRuntimeInformation>
+
+    <PropertyGroup>
+      <Rid Condition=" '$(Rid)' == '' ">$(HostRid)</Rid>
+      <Architecture Condition=" '$(Architecture)' == '' ">x64</Architecture>
+      <OSName Condition=" '$(OSName)' == '' ">$(HostOSName)</OSName>
+
+      <BuildInfoPropsContent>
+&lt;Project ToolsVersion=&quot;15.0&quot;&gt;
+    &lt;PropertyGroup&gt;
+        &lt;Rid&gt;$(Rid)&lt;/Rid&gt;
+        &lt;Architecture&gt;$(Architecture)&lt;/Architecture&gt;
+        &lt;OSName&gt;$(OSName)&lt;/OSName&gt;
+    &lt;/PropertyGroup&gt;
+&lt;/Project&gt;
+      </BuildInfoPropsContent>
+
+      <ExistingBuildInfoPropsContent Condition=" Exists('$(BuildInfoProps)') ">
```

Comment:
This isn't a Boolean to check if it exists. The property loads the contents of the file if it exists. If the file doesn't exist then the content is empty string. So I think the property is named correctly?